extensions/amp-list/0.1/amp-list.js

```diff
@@ -191,6 +222,12 @@ export class AmpList extends AMP.BaseElement {
     );
 
     this.loadMoreEnabled_ = this.element.hasAttribute('load-more');
+    userAssert(
```

Comment:
For my own learning: whats the benefit of having the `userAssert` that load-more and layout=container are incompatible within `buildCallback` vs. `layoutCallback`?

Reply:
In regards to why place the assertion in `buildCallback` as opposed to `layoutCallback`, in this case the intention was to place it as close as possible to the `load-more` attribute being read. If we know it's going to cause issues, we might as well error at the earlier possible opportunity so as not to use more resources. 

And to go deeper, the reason the `load-more` attribute is read in the `buildCallback` is because this is generally where internal state is set in a component instance, including reading attributes.

One might put it in `layoutCallback` if a critical piece of the assertion is costly, dealing with rendering or loading resources.

[Building an AMP Extension](https://github.com/ampproject/amphtml/blob/master/contributing/building-an-amp-extension.md#baseelement-callbacks) is a really helpful resource we have that describes the usage of these callbacks.

re: your other note, yes that's a reasonable assumption for the current state of the code, but I don't think it hurts to be explicit that these are treated as exclusive features, especially if say, in the future, someone else decides to set the internal `loadMoreEnabled_` variable without knowing the implicit dependencies here.

Reply:
Thank you for the thorough explanation and links! Its making me realize that I never got around to reading all of our dev-facing docs after skimming them on my first week 😱. I'll read up on it! Something tells me it will make more sense to me now than it did the last time